tensorflow/compiler/mlir/xla/transforms/map_lhlo_to_scalar_op.h

```diff
@@ -148,44 +163,195 @@ inline Optional<CmpIPredicate> getIntCmpPredicate(
+        .getResult();
+  }
+  if (sourceType.isa<FloatType>() && targetType.isa<FloatType>()) {
+    FloatType src = sourceType.cast<FloatType>();
+    FloatType res = targetType.cast<FloatType>();
+    if (src.getWidth() > res.getWidth()) {
+      return b
+          .create<mlir::FPTruncOp>(lhlo_op.getLoc(), result_types, args,
+                                   mlir::None)
+          .getResult();
+    }
+    if (src.getWidth() < res.getWidth()) {
+      return b
+          .create<mlir::FPExtOp>(lhlo_op.getLoc(), result_types, args,
+                                 mlir::None)
+          .getResult();
+    }
+    // No conversion is needed for the same width floats
+    return args.front();
+  }
+  if (sourceType.isa<IntegerType>() && targetType.isa<IntegerType>()) {
+    IntegerType src = sourceType.cast<IntegerType>();
+    IntegerType res = targetType.cast<IntegerType>();
+    if (src.getWidth() > res.getWidth()) {
+      return b
+          .create<mlir::TruncateIOp>(lhlo_op.getLoc(), result_types, args,
+                                     mlir::None)
+          .getResult();
+    }
+    if (src.getWidth() < res.getWidth()) {
+      return b
+          .create<mlir::ZeroExtendIOp>(lhlo_op.getLoc(), result_types, args,
+                                       mlir::None)
+          .getResult();
+    }
+    // No conversion is needed for the same width integers
+    return args.front();
+  }
+  // TODO: Add other primitive type conversions
+  // if (mlir::FpToSiOp::areCastCompatible(sourceType, targetType)) {
+  //   return b.create<mlir::FpToSiOp>(lhlo_op.getLoc(), result_types,
+  //   args,mlir::None).getResult();
+  // }
+
+  return nullptr;
+}
+
+template <>
+inline Value MapLhloOpToStdScalarOp<xla_lhlo::ExpOp>(
+    xla_lhlo::ExpOp lhlo_op, ArrayRef<Type> result_types, ArrayRef<Value> args,
+    OpBuilder b) {
+  Type element_type = args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    return b
+        .create<::mlir::ExpOp>(lhlo_op.getLoc(), result_types, args, mlir::None)
+        .getResult();
+  }
+  return nullptr;
+}
+
+template <>
+inline Value MapLhloOpToStdScalarOp<xla_lhlo::NegOp>(
+    xla_lhlo::NegOp lhlo_op, ArrayRef<Type> result_types, ArrayRef<Value> args,
+    OpBuilder b) {
+  Type element_type = args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    return b
+        .create<::mlir::NegFOp>(lhlo_op.getLoc(), result_types, args,
+                                mlir::None)
+        .getResult();
+  }
+  return nullptr;
+}
+
+template <>
+inline Value MapLhloOpToStdScalarOp<xla_lhlo::SignOp>(
+    xla_lhlo::SignOp lhlo_op, ArrayRef<Type> result_types, ArrayRef<Value> args,
+    OpBuilder b) {
+  Type element_type = args.front()->getType();
+  if (element_type.isa<FloatType>()) {
+    Value one = b.create<mlir::ConstantOp>(lhlo_op.getLoc(), b.getF32Type(),
+                                           b.getFloatAttr(element_type, 1.0));
+    return b
```

Comment:
I am curious if this return statement needs `.getResult()`. I thought that for ops with one result it is not necessary.